eclipse-theia/theia
# Add eslint rule for localization issues

#### What it does

Adds a new eslint rule that helps developers of Theia to correctly localize the application:

1. Flags calls to `nls.localizeByDefault` which use a default value which does not exist inside of the `nls.metadata.json` file. Once such a call is detected, the rule tries to find the best fitting existing default value using the levenshtein distance.
2. Flags calls to `nls.localize` which could be replaced by `nls.localizeByDefault`.

#### How to test

1. Assert that `yarn lint` does not yield any errors (CI is green)
2. Changing the default value of an existing `localizeByDefault` call slightly should lead to an eslint error and a quickfix is provided.
3. Changing an existing `localizeByDefault(default)` call to `localize(key, default)` should lead to an eslint error and a quickfix is provided here as well.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
